homeassistant/components/samsungtv/config_flow.py

```diff
@@ -169,8 +169,16 @@ async def async_step_import(
         """Handle configuration by yaml file."""
         # We need to import even if we cannot validate
         # since the TV may be off at startup
+        LOGGER.debug("Importing config entry from YAML: %s", user_input)
         await self._async_set_name_host_from_input(user_input)
-        self._async_abort_entries_match({CONF_HOST: self._host})
+        entry, _ = self._async_get_existing_matching_entry()
+        if entry:
+            if self.hass.config_entries.async_update_entry(
+                entry, data={**entry.data, **user_input}
```

Comment:
I don't think we can put the script data in the config entry data, as it can contain a template instance.

Reply:
:( in that case we're back to square one: options flow and/or device trigger
I'll mark it as draft and maybe I will look at it again after #68301 is accepted and merged